src/antelope/wallets/authenticators/MetaKeepAuth.ts

```diff
@@ -0,0 +1,172 @@
+    }
+
+    setEmail(email: string): void {
+        this.accountEmail = email;
+        metakeepCache.setLogged(email);
+    }
+
+    // this is the important instance creation where we define a label to assign to this instance of the authenticator
+    newInstance(label: string): EVMAuthenticator {
+        this.trace('newInstance', label);
+        const auth = new MetaKeepAuth({
+            appId: this.appId,
+            appName: this.appName,
+            reasonCallback: this.reasonCallback,
+        }, label);
+        auth.setEmail(this.accountEmail);
+        return auth;
+    }
+
+    async assertMetakeepSDKReady(): Promise<void> {
+        if (!metakeep) {
+            const chainSettings = (useChainStore().getChain(CURRENT_CONTEXT).settings as EVMChainSettings);
+            const endpoint = chainSettings.getRPCEndpoint();
+            const url = `${endpoint.protocol}://${endpoint.host}:${endpoint.port}${endpoint.path ?? ''}`;
+
+            const chainId: number = parseInt(chainSettings.getChainId());
+
+
+            const rpcNodeUrls = {
+                [chainId]: url,
+            } as unknown as Map<number, string>;
+
+            metakeep = new MetaKeep({
+                // App id to configure UI
+                appId: this.appId,
+                // Default chain to use
+                chainId,
+                // RPC node urls map
+                rpcNodeUrls,
+                // Signed in user's email address
+                user: {
+                    email: this.accountEmail,
+                },
+            });
+
+            const provider = await metakeep.ethereum;
+            await provider.enable();
+            web3Provider = new ethers.providers.Web3Provider(provider);
+        }
+    }
+
+    async login(network: string): Promise<addressString | null> {
+        this.trace('login', network);
+        useFeedbackStore().setLoading(`${this.getName()}.login`);
+
+        await this.assertMetakeepSDKReady();
+
+        // metakeepCache
+        const accountAddress = metakeepCache.getEthAddress(this.accountEmail);
+        if (accountAddress) {
+            this.accountAddress = accountAddress;
+            useFeedbackStore().unsetLoading(`${this.getName()}.login`);
+            return this.accountAddress as addressString;
+        }
+
+
+        const provider = await this.web3Provider();
+        this.accountAddress = await provider.getSigner().getAddress();
+
+        useFeedbackStore().unsetLoading(`${this.getName()}.login`);
+
+
+        const credentials = await metakeep?.getWallet();
+        metakeepCache.addCredentials(this.accountEmail, credentials.wallet);
+
+        return this.accountAddress as addressString;
+    }
+
+    async logout(): Promise<void> {
+        this.trace('logout');
+        metakeepCache.setLogged('');
+        return Promise.resolve();
+    }
+
+    async isConnectedTo(chainId: string): Promise<boolean> {
+        this.trace('isConnectedTo', chainId);
+        return true;
```

Comment:
is it not possible to check chainID with metakeep?

Reply:
by default, they support any EOS, ETH, or SOL-compatible blockchains.

This is a good topic for a KT I think.